lib/core/TokenSet.ts

```diff
@@ -0,0 +1,55 @@
+import { OktaAuthCoreInterface } from './types';
+import { TokenManager } from '../oidc';
+import { TOKEN_STORAGE_NAME } from '../constants';
+import { ServiceManager } from './ServiceManager';
+
+export class TokenSet {
+  private sdk: OktaAuthCoreInterface;
+  private map: Map<string, any>; // TODO: add type
+
+  constructor(sdk: OktaAuthCoreInterface) {
+    this.sdk = sdk;
+
+    this.map = new Map();
+    const options = sdk.options;
+    const tokenManager = new TokenManager(this.sdk, options.tokenManager);
+    const serviceManager = new ServiceManager(this.sdk, options.services);
+    this.map.set(TOKEN_STORAGE_NAME, {
+      tokenManager,
+      serviceManager,
+    });
+  }
+
+  start(key?: string) {
+    // start services for one or all token sets
+  }
+
+  stop(key?: string) {
+    // stop services for one or all token sets
+  }
+
+  // call this method when need to add a new token set, like after receiving new tokens from authorize/idx
+  addTokenSet(tokens, options) {
+    const storageKey = options.storageKey;
+    const tokenManager = new TokenManager(this.sdk, options.tokenManager); // with specific storageKey
+    const serviceManager = new ServiceManager(this.sdk, options.services);
+    this.map.set(storageKey, {
+      tokenManager,
+      serviceManager,
+    });
+
+    tokenManager.setTokens(tokens);
+
+    if (options.default) {
+      this.useTokenSet(options.tokenManager.storageKey);
+    }
+  }
+
+  useTokenSet(key) {
```

Comment:
We will need to make sure each query end up with only one token set, so maybe just use one type of identifier. For internal api (myaccount), we can find the best practice to pick the key, otherwise it leaves to devs to figure out the key.